Revert "refactoring proposal : turn Stage into a context manager"

tgamblin opened the pull request:
Reverts LLNL/spack#477

Reverting this while I fix a few bugs.  Merged too early!

alalazo commented:
@tgamblin How does it work with reverted merges? Should I restore the original branch and fix something there?

tgamblin commented:
@alalazo: I fiddled with this a bit more and got it to where it worked.  I think the bug was that it would fail on `spack fetch` and `spack stage` because the stage wasn't created until the `with self.stage` in `do_install()`.  I put the explicit call to `Stage.create` back in `_make_stage` and that fixed things.  I also simplified the `do_install` code a bit and updated the Stage documentation slightly.  See what you think -- it's merged again now as of e515042.
